### ISSUE TITLE
patronictl failing to list members or perform commands needing cluster name

### ISSUE THREAD

**Issue by bhazuka** [zalando/patroni#477]
OS: CentOS 7
etcd version: 3.2.1
patroni version 1.2.5
postgresql version: 9.6.3

When running patronictl commands that interrogate the cluster, the following error occurs:
_*Error: 'Can not find suitable configuration of distributed configuration store\nAvailable implementations: '*_

Note that patroni appears to be running and monitoring the cluster properly and curl commands do return data as seen below.
```
# curl http://localhost:2379/v2/keys/pgHAdev7/ts2/members/ | jq
{
  "action": "get",
  "node": {
    "key": "/pgHAdev7/ts2/members",
    "dir": true,
    "nodes": [
      {
        "key": "/pgHAdev7/ts2/members/postgresql_104",
        "value": "{\"role\":\"master\",\"state\":\"running\",\"conn_url\":\"postgres://192.168.1.104:5432/postgres\",\"api_url\":\"https://192.168.1.104:8008/patroni\",\"xlog_location\":33555144}",
        "expiration": "2017-07-15T23:19:48.885312564Z",
        "ttl": 28,
        "modifiedIndex": 38312,
        "createdIndex": 38312
      }
    ],
    "modifiedIndex": 37222,
    "createdIndex": 37222
  }
}

```

Here is the configuration information from patroni that I feel pertains to the problem.
```
namespace: /pgHAdev7/
scope: ts2
name: postgresql_104

etcd:
    host: 192.168.1.104:2379
    url: "http://localhost:2379"
```

Here is the configuration information from patronictl after performing `patronictl configure`.
```
{dcs_api: 'etcd://localhost:2379', namespace: '/pgHAdev7/'}
```

**Comment by CyberDem0n** [comment]
Hi @bhazuka ,

This is all very strange...
How did you installed Patroni?
Where is your patronictl located?
Is patroni installed on the host where you trying to run patronictl?

And a few words regarding configuration:
1. you don't need to specify `etcd.host` and `etcd.url` at the same time, you can just use one of them, for example `etcd.url`
2. patronictl can use the same config file as Patroni itself.

**Comment by bhazuka** [comment]
I think I figured it out, well partially anyway. The patronictl I called was generated via pyinstaller.
`pyinstaller --clean --log-level=INFO --onefile patronictl.py` If I pull the v1.2.5 tar and create a virtual environment and point to the patroni configuration then it appears to work fine. So, the issue appears to be with the build from pyinstaller. Pyinstaller must require some additional configuration to compile patronictl into a single binary. I am not well versed in pyinstaller so I am not really sure what additional configuration I will need to perform to get patronictl to compile to a single binary for distribution.
